tests/unit/models/transactions/test_set_oracle.py

```diff
@@ -0,0 +1,216 @@
+import time
+from unittest import TestCase
+
+from xrpl.models.exceptions import XRPLModelException
+from xrpl.models.transactions import OracleSet
+from xrpl.models.transactions.oracle_set import (
+    MAX_ORACLE_PROVIDER,
+    MAX_ORACLE_SYMBOL_CLASS,
+    MAX_ORACLE_URI,
+    PriceData,
+)
+
+_ACCOUNT = "rsA2LpzuawewSBQXkiju3YQTMzW13pAAdW"
+_PROVIDER = "chainlink"
+_ASSET_CLASS = "currency"
+
+_EMPTY_PROVIDER = ""
+_LENGTHY_PROVIDER = "X" * (MAX_ORACLE_PROVIDER + 1)
+
+
+class TestSetOracle(TestCase):
+    def test_valid(self):
+        tx = OracleSet(
+            account=_ACCOUNT,
+            oracle_document_id=1,
+            provider=_PROVIDER,
+            asset_class=_ASSET_CLASS,
+            last_update_time=int(time.time()),
+            price_data_series=[
+                PriceData(
+                    base_asset="XRP", quote_asset="USD", asset_price=740, scale=1
+                ),
+                PriceData(
+                    base_asset="BTC", quote_asset="EUR", asset_price=100, scale=2
+                ),
+            ],
+        )
+        self.assertTrue(tx.is_valid())
+
+    def test_missing_data_series(self):
+        with self.assertRaises(XRPLModelException):
+            OracleSet(
+                account=_ACCOUNT,
+                oracle_document_id=1,
+                provider=_PROVIDER,
+                asset_class=_ASSET_CLASS,
+                last_update_time=int(time.time()),
+            )
```

Comment:
Can you also verify the exception that's being thrown is the intended one? You can refer to the `test_amm_create.py` example where it uses code to assert the error message:
```python
self.assertEqual(
                error.exception.args[0],
                "{'trading_fee': 'Must be between 0 and 1000'}",
            )
```

Reply:
updated in [72afddf](https://github.com/XRPLF/xrpl-py/pull/701/commits/72afddfcad7ece7e236653c3229c1416859a3f93)

We are not validating the exceptions and error messages in the correct manner, in the current state of the codebase. After an exception is throw, the `unittest.TestCase` framework does not verify any of the `self.assertEqual` statements. Please check this for an example: https://github.com/XRPLF/xrpl-py/commit/abc3b0cff17a9a1566d531c3a94511ccbfa702f6

We need to bring the `self.assertEqual` statements outside the context of the `self.assertRaises` and later perform validations.